### PR TITLE
Create index db & commit index entries

### PR DESCRIPTION
**Pull request by cendhu**

### CODE REVIEW

File: internal/blockprocessor/committer.go [hunk 1]
```diff
@@ -360,19 +373,42 @@ func constructDBEntriesForDBAdminTx(tx *types.DBAdministrationTx, version *types
...
+
+		// Only when the index definition is provided for the existing DB for the
+		// first time, we will reach here
+		indexDB := &worldstate.KVWithMetadata{
+			Key: stateindex.IndexDBPrefix + dbName,
+			Metadata: &types.Metadata{
+				Version: version,
+			},
+		}
+		indexForExistingDBs = append(indexForExistingDBs, indexDB)
```
Comment (tock-ibm) [inline internal/blockprocessor/committer.go:406]: This loop is pretty difficult to read. I suggest to create to booleans that capture the two conditions:
`indexIsDefined := dbIndex != nil && dbIndex.GetAttributeAndType() != nil`
and
`indexAlreadyExists := db.Exist(stateindex.IndexDBPrefix + dbName)`
and then use them and their converse in switch or if-else statements, that will make it clear what you are trying to achieve and make it easier to read and instantly grasp what happens in each case (there are 4, but maybe on is degenerate?).

File: internal/blockprocessor/committer_test.go [hunk 2]
```diff
@@ -845,16 +846,19 @@ func TestStateDBCommitterForDBBlock(t *testing.T) {
...
-		expectedDBsAfter  []string
-		expectedIndex     map[string]*types.DBIndex
-		valInfo           []*types.ValidationInfo
+		name                    string
+		setup                   func(db worldstate.DB)
+		expectedDBsBefore       []string
+		expectedIndexDBsBefore  []string
+		dbAdminTx               *types.DBAdministrationTx
+		expectedDBsAfter        []string
+		expectedIndex           map[string]*types.DBIndex
```
Comment (tock-ibm) [inline internal/blockprocessor/committer_test.go:855]: In line with the naming convention around, `expectedIndex` sould be `expectedIndexAfter` no?

And, in addition, in the spirit of OCD, I noticed there is no `expectedIndexBefore`...

**Comment by tock-ibm** [comment]
It looks good except some readability suggestions.

File: internal/blockprocessor/committer_test.go (hunk 2 above)
Comment (cendhu) [reply internal/blockprocessor/committer_test.go]: Done.

File: internal/blockprocessor/committer.go (hunk 1 above)
Comment (cendhu) [reply internal/blockprocessor/committer.go]: Agree. Hope, I have simplified it.

File: internal/blockprocessor/committer.go [hunk 3]
```diff
@@ -337,21 +338,33 @@ func constructDBEntriesForDBAdminTx(tx *types.DBAdministrationTx, version *types
 			}
 			value = v
 
+			// for each DB, if index is defined, we need to create an
+			// index DB to store index entries for that DB
```
[the remark this answers is not in the document]
Comment (cendhu) [reply internal/blockprocessor/committer.go:367]: Done.

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**